src/gui/creds/webflowcredentialsdialog.cpp

```diff
@@ -62,8 +62,10 @@ WebFlowCredentialsDialog::WebFlowCredentialsDialog(Account *account, bool useFlo
         connect(_webView, &WebView::urlCatched, this, &WebFlowCredentialsDialog::urlCatched);
     }
 
-    auto app = static_cast<Application *>(qApp);
-    connect(app, &Application::isShowingSettingsDialog, this, &WebFlowCredentialsDialog::slotShowSettingsDialog);
+    connect(ownCloudGui::instance(), &ownCloudGui::isShowingSettingsDialog, this, &WebFlowCredentialsDialog::slotShowSettingsDialog);
+
+    // Dialog visibility
+    connect(this, &WebFlowCredentialsDialog::onSetVisible, ownCloudGui::instance(), &ownCloudGui::slotDialogVisibilityChanged);
```

Comment:
I wonder if we want that coupling... if we need ownCloudGui to monitor states from some dialogs, why not having an event filter for that and register those dialogs to it? Properly done ownCloudGui might need to know only about that event filter class and the dialogs might not even need to know about it.